### ISSUE TITLE
Prisma.yml `schema` property definition? (and how does .graphqlconfig.yml work)

### ISSUE THREAD

**Issue by lrholmes** [prismagraphql/prisma#1824]
I'm a little confused about how the `schema` property works in the `prisma.yml` file, but also how the file works more generally in conjunction with `.graphqlconfig.yml`.

First of all, defining the `schema` property in `prisma.yml` appears to do nothing. In order to change the location where my generated schema goes, I had to change the `database:schemaPath` value in `.graphqlconfig.yml`, and then `prisma deploy` worked. I found this odd considering the [Prisma documentation](https://www.prismagraphql.com/docs/reference/service-configuration/prisma.yml/yaml-structure-ufeshusai8#schema-(optional)) for the `schema` property makes no mention of the `.graphqlconfig.yml` file.

So that's one issue, but this made me realise that I don't really understand what the `.graphqlconfig.yml` file actually does, and how it works with Prisma. I have it defined because it was part of an example project I looked at during setup. If I remove it, I can't seem to deploy anymore (no schema is generated, anyway) so it seems to be required, but I can't find any docs to suggest what fields in this file are actually consumed by Prisma.

Any clarification on that would be great - thanks.

**Comment by johannpinson** [comment]
Hey @lrholmes 

I have the same problem just today, when I'm trying to modify the default boilerplate structure.

This is the actual line which solves you when you run a `prisma deploy`:
https://github.com/graphcool/prisma/blob/master/cli/packages/prisma-cli-core/src/commands/deploy/index.ts#L484

Actually, the Prisma CLI use `graphql-cli` inside it, and after a deploy you can see inside your console a `graphql prepare` to generate the schema output.

&nbsp;
Basically, it's the `schema` property of the `prisma.yml` which is used.
You can considerate it as the minimal setting for a simple export!

But the `graphqlconfig.yml` is a config file for the `graphql-cli`, it's this file which is selected to get the advanced configuration of your schema output.


Hope that's my feedback can help you 😉

**Comment by steve-a-jones** [comment]
This has been a pain point for me as well -- it is fairly difficult to _reason about_ how .graphqlconfig is being used.

For instance.. the name of the keys under `project` can be arbitrary as this logic https://github.com/graphcool/prisma/blob/670f490fa25cecc7c4b279c2ace945a232492ea4/cli/packages/prisma-cli-core/src/commands/deploy/getSchemaPathFromConfig.ts#L13 just finds a "prisma"ish configuration.

**Comment by nikolasburk** [comment]
Thanks a lot for opening this issue @lrholmes! The `schema` property has been removed from prisma.yml, more info in our [new docs](https://docs-beta.prisma.io/1.13/develop-prisma-service/service-configuration/prisma-yml-5cy7/).
